File: laravel-demo/app/Http/Controllers/Pages/MessagesV3Controller.php
<?php

namespace App\Http\Controllers\Pages;

use App\User;
use App\Http\Controllers\Controller;

class MessagesV3Controller extends Controller
{
    public function compose()
    {
        return view('pages.messages-v3.compose', ['title' => 'Compose message v3 - Pages']);
    }

    public function item()
    {
        return view('pages.messages-v3.item', ['title' => 'Message item v3 - Pages']);
    }

    public function list()
    {
        return view('pages.messages-v3.list', ['title' => 'Messages list v3 - Pages']);
    }

}

File: rails-demo/app/controllers/pages/messages_v3_controller.rb
class Pages::MessagesV3Controller < ApplicationController
  def compose
    @title = 'Compose message v3 - Pages'
  end

  def item
    @title = 'Message item v3 - Pages'
  end

  def list
    @title = 'Messages list v3 - Pages'
  end

end
